Sources/core/objects/Gun.cpp

```diff
@@ -0,0 +1,96 @@
+#include "Gun.h"
+#include "Tile.h"
+
+Gun::Gun(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+    ammunition_ = 1;
+    max_ammunition_ = 1;
+}
+
+bool Gun::UseAmmo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Gun::Shoot(VDir target)
+{
+    id_ptr_on<CubeTile> tile = GetOwner()->GetOwner();
+    if (tile.valid())
+    {	
+        if(UseAmmo())
+        { 
+            id_ptr_on<Projectile> p = Create<Projectile>(Bullet::T_ITEM_S(),GetOwner()->GetOwner());
+            p->MakeMovementPattern(target,GetOwner());	
+        }
+        else
+        {
+            PlaySoundIfVisible("empty.ogg",owner.ret_id());
+		// the *click* text from ss13
+        }
+    }	
+}
+
+bool Gun::AddAmmo()
+{
+    if (ammunition_ < max_ammunition_)
+    {
+        ++ammunition_;
+        return true;
+    }
+    return false;
+}
+void Gun::AttackBy(id_ptr_on<Item> item)
+{
+    if (id_ptr_on<AmmunitionBox> r = item)
+    {
```

Comment:
Again, instead of `r` it should be some word (for example, `box`, because `ammunition_box` will be too long)